pkg/expr/expr.go

```diff
@@ -150,9 +151,14 @@ func newConstant(v string, vType config.ValueType) (*Constant, error) {
 			return nil, err
 		}
 	default: // string
```

Comment:
perhaps this comment will help.

// newConstant converts literals recognized by parser (int, float, char, string) to config.ValueType

Several config.ValueTypes are "strings", so they must be parsed separately as those value types, and then the correct vType should be set.